### PR TITLE
Tools: Husky upgrade and git ignorance improved

### PR DESCRIPTION
**Pull request by shivapoudel**
## Description

- With husky 7.0.0, .husky/.gitignore is now unnecessary and can be removed. Ref: https://github.com/WordPress/gutenberg/pull/32077#issuecomment-855802667
- Add declaration to PHP files that will always have LF line endings on checkout.
- Windows-specific file `Thumbs.db` ignorance.
- Remove composer created vendor dir by including in `.gitignore`.

## Types of changes

Just a development files changes.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,7 +1,10 @@
-# default
+# Set the default behavior, in case people don't have `core.autocrlf` set.
 * text=auto
 
-# Windows bat files in mobile/android
+# Declare files that will always have LF line endings on checkout.
+*.php text eol=lf
```
Comment (gziolo) [inline .gitattributes]: Can you explain this particular change. It isn't clear why PHP files would need special treatment. It also seems completely unrelated to the Husky upgrade.

**Review by gziolo — COMMENTED** [review]
I have only one concern regarding the changes in `.gitattributes`. Otherwise it looks like Husky v7 fixed the issue with `_` subfolder 👍🏻 

Thank you for the follow-up on the Husky integration 😄

File: .gitattributes (hunk 1 above)
Comment (shivapoudel) [reply .gitattributes]: @gziolo This is useful for users on Windows OS who don't have `core.autocrlf` set. Yeah, it seems a bit unrelated to the Husky upgrade but I have attached this small change as I figured out while touching the PHP files on Gutenberg. More over this was derived from the WooCommerce project with this raised PR https://github.com/woocommerce/woocommerce/pull/27456

Comment (gziolo) [reply .gitattributes]: I would prefer to have this change proposed as a separate PR. I tested everything else and it works as expected.

**Review by gziolo — APPROVED** [review]
Nice one, thank you for your contribution 💯

**Comment by gziolo** [comment]
It looks like `vendor/.gitignore` ensures that the `vendor` folder exists while running the script that builds the zip file for the Gutenberg plugin:

Screen Shot 2021-07-06 at 11 27 19

There might be more places related to vendor scripts that assume that the `vendor` folder exists.

It's very unfortunate, but I think it's simpler to leave it as is for now to unblock this PR.

**Review by gziolo — COMMENTED** [review]
See above.

**Comment by shivapoudel** [comment]
@nylen As I am aware of `vendor` directory is utilized by Composer dependencies but it is used to download vendor JS files through this PR https://github.com/WordPress/gutenberg/pull/1025

@gziolo If this vendor JS files can be in `tmp`, `dist` or somewhere else that would be nice in my opinion!

**Review by gziolo — APPROVED** [review]
All green, let's land it 🎉

**Comment by gziolo** [comment]
> If this vendor JS files can be in tmp, dist or somewhere else that would be nice in my opinion!

Good observation, maybe `build/vendor/` would be a better location, but it requires some changes in how those scripts are downloaded and consumed.
